src/backend/distributed/commands/citus_add_local_table_to_metadata.c

```diff
@@ -333,6 +350,38 @@ CreateCitusLocalTable(Oid relationId, bool cascadeViaForeignKeys)
 }
 
 
+void
+CreatePartitionedCitusLocalTable(Oid parentOid, bool cascadeViaForeignKeys)
```

Comment:
I think we should avoid a special function `CreatePartitionedCitusLocalTable`.

All tables should simply be created with `CreateCitusLocalTable`.

The idea should be in 
1) As we already do today, we'll drop all the foreign keys (including parents) in CascadeOperationForConnectedRelations
2)  We'll later detach partitions in `CascadeOperationForConnectedRelations
3) At this point, we'll have regular local tables without partitioning or foreign keys
4) We'll call `ExecuteCascadeOperationForRelationIdList` so that each table is converted to citus local table via `CreateCitusLocalTable`
5) We'll attach partitions back
6) We'll create foreign keys on the parents (which will cascade into partitions automatically)

Wouldn't the above solve our problems in a simpler manner?


As we discussed, this approach limits the arbitrary foreign keys such as a single partition having a foreign key to another table but other partitions do not (e.g., we already have `ErrorIfAnyPartitionRelationInvolvedInNonInheritedFKey `). So, that is acceptable for the first iteration.